gapis/api/vulkan/state_rebuilder.go

```diff
@@ -1498,20 +1501,24 @@ func (sb *stateBuilder) createImage(img ImageObjectʳ, imgPrimer *imagePrimer) {
 	} else {
 		// Otherwise, we have no sparse bindings, we are either non-sparse, or empty.
 		if !isDenseBound(img) {
-			return
+			return nil, fmt.Errorf("no sparse bindings found")
 		}
 		walkImageSubresourceRange(sb, img, sb.imageWholeSubresourceRange(img), appendImageLevelToOpaqueRanges)
 		// TODO: Handle multi-planar images
-		planeMemInfo, _ := subGetImagePlaneMemoryInfo(sb.ctx, nil, api.CmdNoID, nil, sb.oldState, GetState(sb.oldState), 0, nil, nil, img, VkImageAspectFlagBits(0))
-		vkBindImageMemory(sb, img.Device(), img.VulkanHandle(),
+		planeMemInfo, _ := subGetImagePlaneMemoryInfo(sb.ctx, nil, api.CmdNoID, nil, srcState, GetState(srcState), 0, nil, nil, img, VkImageAspectFlagBits(0))
+		vkBindImageMemory(sb, img.Device(), vkImage,
 			planeMemInfo.BoundMemory().VulkanHandle(), planeMemInfo.BoundMemoryOffset())
 	}
 	// opaqueRanges should contain all the bound image subresources by now.
 	if len(opaqueRanges) == 0 {
 		// There is no valid data in this image at all
-		return
+		return nil, fmt.Errorf("no valid data in the image")
```

Comment:
I think same as before, this might? not actually be an error condition. Please double-check the spec.